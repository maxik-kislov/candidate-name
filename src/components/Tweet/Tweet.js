import React from 'react';
import PropTypes from 'prop-types';
import './Tweet.scss';

const Tweet = ({ username, text, image }) => (
  <li className="item">
    <div className="item__image">
      <img className="item__img" src={image} alt={username} />
    </div>
    <div className="item__body">
      <span className="item__username">{username}</span>
      <span className="item__text">{text}</span>
    </div>
  </li>
);

Tweet.defaultProps = {
  username: '',
  text: '',
  image: '',
};

Tweet.propTypes = {
  username: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
};

export default Tweet;
