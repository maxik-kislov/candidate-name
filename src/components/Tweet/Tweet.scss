.item {
  background: #fff;
  border-radius: 4px;
  -webkit-box-shadow: 0 1px 2px rgba(0, 0, 0, 0.2);
  box-shadow: 0 1px 2px rgba(0, 0, 0, 0.2);
  padding: 16px;
  margin-bottom: 20px;
  display: flex;
  flex-direction: row;

  &:hover {
    background-color: #e7f4fe;
  }

  &__image {
    border-radius: 64px;
    overflow: hidden;
    width: 64px;
    height: 64px;
  }

  &__img {
    display: block;
    width: 100%;
  }

  &__body {
    flex: 1;
    padding-left: 16px;
  }

  &__username {
    display: block;
    font-weight: bold;
    margin-bottom: 8px;
  }

  &__text {
  }
}
