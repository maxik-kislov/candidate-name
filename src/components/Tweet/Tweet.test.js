import setupTests from '../../setupTests';
import React from 'react';
import { shallow } from 'enzyme';
import Tweet from './Tweet';

const render = props => shallow(<Tweet {...props} />);

describe('Tweet', () => {
  it('should contain an element with a class .item', () => {
    const element = render().find('.item');
    expect(element.exists()).toBeTruthy();
  });

  it('should render correctly', () => {
    expect(render().getElements()).toMatchSnapshot();
  });
});
