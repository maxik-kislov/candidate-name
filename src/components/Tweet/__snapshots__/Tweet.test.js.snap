// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tweet should render correctly 1`] = `
Array [
  <li
    className="item"
  >
    <div
      className="item__image"
    >
      <img
        alt=""
        className="item__img"
        src=""
      />
    </div>
    <div
      className="item__body"
    >
      <span
        className="item__username"
      >
        
      </span>
      <span
        className="item__text"
      >
        
      </span>
    </div>
  </li>,
]
`;
